cmd/util/version.go

```diff
@@ -65,23 +69,82 @@ func GetAllVersions(ctx context.Context) (Versions, error) {
 
 var printMessage *string = nil
 
+const updateStatePath string = "update.json"
+const updateFrequency time.Duration = 24 * time.Hour
+
 // StartUpdateCheck is a Cobra pre run hook to run an update check in the
 // background. There should be no output if the check fails or the context is
 // cancelled before the check can complete.
 func StartUpdateCheck(cmd *cobra.Command, args []string) {
```

Comment:
We could pass the repo path as an argument to this method and remove the method `GetFSRepo` which looks in the context.

Reply:
Alternatively, if that is not possible, lets un-export the method `GetFSRepo` and it's internals as the pattern shouldn't be used outside this specific case.

Reply:
The challenge is that this method is implementing the pre/post-run hooks interface, which only accepts string arguments and a context. AFAICS the context is the only way to pass state into hooks, without using a global variable which I think would be worse.

I think un-exporting the method is a sensbile change, and I'll add a comment to the effect of "don't start using this because context is intended to be used sparingly".

Reply:
It turns out using the config system properly removes the need to do any of this, as it will now set up the absolute path on repo initialisation rather than exposing the repo here. Nice.